deepmd/descriptor/se_a.py

```diff
@@ -542,6 +555,20 @@ def build(
                 dtype=GLOBAL_TF_FLOAT_PRECISION,
             )
             t_ntypes = tf.constant(self.ntypes, name="ntypes", dtype=tf.int32)
+            t_ntypes_spin = tf.constant(
+                self.ntypes_spin, name="ntypes_spin", dtype=tf.int32
+            )
+            if self.spin is not None:
+                t_virtual_len = tf.constant(
+                    self.spin.virtual_len,
+                    name="virtual_len",
+                    dtype=GLOBAL_TF_FLOAT_PRECISION,
+                )
+                t_spin_norm = tf.constant(
+                    self.spin.spin_norm,
+                    name="spin_norm",
+                    dtype=GLOBAL_TF_FLOAT_PRECISION,
+                )
```

Comment:
These codes are not directly related to the descriptor.
I suggest moving these lines to `Spin.build`, and call `Spin.build` in `EnerModel`.